sql/core.go

```diff
@@ -399,6 +399,35 @@ type ProjectedTable interface {
 	Projections() []string
 }
 
+// TODO: lots of stuff
+type Bucket interface {
+	SetValue(float64)
+	SetFrequency(float64)
+
+	IsSingleton() bool
+	GetValue() float64
+	GetFrequency() float64
+}
+
+type Histogram map[float64]Bucket
+
+type ColumnStatistic interface {
+	GetMean() float64
+	GetMin() float64
+	GetMax() float64
+	GetCount() uint64
+	GetNullCount() uint64
+	GetBucket(value float64) Bucket
+	GetHistogram() Histogram
+}
+
+type ColumnStatisticsMap map[string]ColumnStatistic
+
+type Statistics interface {
```

Comment:
is there a reason these need to be interface? seems like they could just be a struct